jib-core/src/main/java/com/google/cloud/tools/jib/http/Connection.java

```diff
@@ -45,15 +47,43 @@
  */
 public class Connection implements Closeable {
 
-  /**
-   * Do not use {@link NetHttpTransport}. It does not process response errors properly. A new {@link
-   * ApacheHttpTransport} needs to be created for each connection because otherwise HTTP connection
-   * persistence causes the connection to throw {@link NoHttpResponseException}.
-   *
-   * @see <a
-   *     href="https://github.com/google/google-http-java-client/issues/39">https://github.com/google/google-http-java-client/issues/39</a>
-   */
-  private HttpRequestFactory requestFactory = new ApacheHttpTransport().createRequestFactory();
+  public static class Builder {
+
+    private final URL url;
+
+    /**
+     * Do not use {@link NetHttpTransport}. It does not process response errors properly. A new
+     * {@link ApacheHttpTransport} needs to be created for each connection because otherwise HTTP
+     * connection persistence causes the connection to throw {@link NoHttpResponseException}.
+     *
+     * @see <a
+     *     href="https://github.com/google/google-http-java-client/issues/39">https://github.com/google/google-http-java-client/issues/39</a>
+     */
+    private HttpTransport httpTransport = new ApacheHttpTransport();
+
+    public Builder(URL url) {
+      this.url = url;
+    }
+
+    /**
+     * Turns off the normal TLS peer verification.
+     *
+     * @throws GeneralSecurityException if unable to turn off
+     * @return this
+     */
+    public Builder doNotValidateCertificate() throws GeneralSecurityException {
+      ApacheHttpTransport.Builder transportBuilder = new ApacheHttpTransport.Builder();
```

Comment:
I think it would be fine to have this as one line `httpTransport = new ApacheHttpTransport.Builder().doNotValidateCertificate().build();`